imports/api/classes/Server.jsx

```diff
@@ -613,16 +615,46 @@ export default class Server {
     }
 
     smppReceive(from, to, message) {
-        //this confirms that a new sms was received
-
-        //TODO: 
-        //get owner of the did number (didOwner function)
-        //get number of parts the message body has to calculate total sms price (use getParts)
-        //check if account is billable, if not set result to insufficient funds
-        //get application related to the did number (didApp function)
-        //construct object to be sent to the msg_url of the application
-        //charge the account (use manager class when inserting record)
-        //send the constructed object to msg_url using processRequestUrl
+        let owner = this.didOwner(from);
+        const Message = new MessageManager(owner.account_id);
+        Message.parseJSON({
+            from: from,
+            to: to,
+            body: message
+        });
+        let price = Message.getPrice();
+        let query = "SELECT * FROM `accounts` WHERE `account_id`=?";
+        let result = this.dbConnection.selectOne(query, [owner.account_id]);
+        if (result) {
+            this.accountData = result;
+            if(!this.isAccountBillable(result, price))
+                return {
+                    success: false,
+                    code: 400,
+                    error: 'Insufficient funds'
+                };
+            const app = this.didApp(owner.account_id, from);
+            if(app)
+                this.processRequestUrl(Message.json, app.msg_url, app.msg_method, app.msg_fb_url, app.msg_fb_method);
```

Comment:
Do not send raw database data to the callback url. Generate a new one, refer to old source code